sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,399 @@
+
+  // Get
+
+  // return marray of component
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  // return sequence of elements of component
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<T, int_seq::size()> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<T, int_seq::size()> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Set
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The class `sycl::ext::oneapi::marray<sycl::ext::oneapi::complex<T>, N>`,
+has instantiations of `T`; `float`, `double`, and `sycl::half` declared.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<double>, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<float>, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<sycl::half>, NumElements>;
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The generic type `mgencomplex` is defined as types
+`marray<sycl::ext::oneapi::complex<double>, {N}>`,
+`marray<sycl::ext::oneapi::complex<float>, {N}>`,
+`marray<sycl::ext::oneapi::complex<sycl::half>, {N}>`.
+
+The table below shows the free functions operating on the `marray` complex
+class. No table is provided for the `marray` class as no changes to it are
+proposed.
+
+[%header,cols="5,5"]
+|===
+|Function
+|Description
+
+|`mgencomplex make_complex_marray(const mgenfloat& x, const mgenfloat& y);`
+|Constructs a marray of complex numbers with real values in marray x, and the imaginary values in marray y.
+|`mgencomplex make_complex_marray(const mgenfloat& x, const genfloat& y);`
```

Comment:
Just checking ... is it your intention to pass this scalar value `y` by reference rather than by value?

Reply:
That is a good catch. I have removed `genfloat` from being passed by reference to this and the `set_X` free functions.